bot/exts/fun/hangman.py

```diff
@@ -0,0 +1,181 @@
+from asyncio import TimeoutError
+from pathlib import Path
+from random import choice
+from typing import Literal
+
+from discord import Embed, Message
+from discord.ext import commands
+
+from bot.bot import Bot
+from bot.constants import Colours, NEGATIVE_REPLIES
+
+# Defining all words in the list of words as a global variable
+with Path("bot/resources/fun/hangman_words.txt").resolve().open(mode="r", encoding="utf-8") as f:
+    ALL_WORDS = [line.strip('\n') for line in f.readlines()]
+
+# Defining a list of images that will be used for the game to represent the hangman person
+IMAGES = {
+    6: "https://cdn.discordapp.com/attachments/859123972884922418/887067234760032296/hangman0.png",
+    5: "https://cdn.discordapp.com/attachments/859123972884922418/887067240204206220/hangman1.png",
+    4: "https://cdn.discordapp.com/attachments/859123972884922418/887067244578881626/hangman2.png",
+    3: "https://cdn.discordapp.com/attachments/859123972884922418/887067248546685029/hangman3.png",
+    2: "https://cdn.discordapp.com/attachments/859123972884922418/887067252099268608/hangman4.png",
+    1: "https://cdn.discordapp.com/attachments/859123972884922418/887067255429554197/hangman5.png",
+    0: "https://cdn.discordapp.com/attachments/859123972884922418/887067260726939728/hangman6.png",
+}
```

Comment:
Wouldn't it be smarter to store these in the resources folder, since doing it this way makes it reliant on the channel and messages existing?

Reply:
I did that in the beginning, though the game became quite buggy and slow -- as it had to load from there every time due to a closing file operation from discord.py, I think the current way is likely the best for the game

Reply:
GitHub serves the file over http, which you can use (which was my original intention) like this: https://raw.githubusercontent.com/python-discord/sir-lancebot/main/bot/resources/fun/all_cards.png

Reply:
> GitHub serves the file over http, which you can use (which was my original intention) like this: https://raw.githubusercontent.com/python-discord/sir-lancebot/main/bot/resources/fun/all_cards.png

Ah, I'll change it to that then